src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -1,60 +1,117 @@
 package co.inventorsoft.academy.collections.model;
 
 import java.util.Collection;
+import java.util.HashSet;
 import java.util.Iterator;
 import java.util.Set;
 
 public class Range<T> implements Set<T> {
 
+    private HashSet<T> range;
+    private static final Range<Byte> rByte = new Range();
+    private static final Range<Short> rShort = new Range();
+    private static final Range<Integer> rInt = new Range();
```

Comment:
You have to be able to add arbitrary number of types, you don't have to create all of them here.
Let's say I'm going to create a new numeric type that has only odd number, you should be able to handle it without changing your code.